pages/projects/index.js

```diff
@@ -1,77 +1,87 @@
+import { fetchAllStrapiProjects } from "../../lib/strapi";
+import { ProjectCard } from "../../components/projectSpotlight";
+import { SearchBar } from "@/components/projects/search-bar";
 
-export default function Projects({ projects, size = 'medium' }) {
-  const theme = useTheme();  
+const filterData = (query, data) => {
+  if (!query) {
+    return data;
+  } else {
+    return data.filter((d) => d.name.toLowerCase().includes(query));
+  }
+};
+
+export default function Projects({ projects, size = "medium" }) {
+  const theme = useTheme();
+  const [searchQuery, setSearchQuery] = useState("");
+  const projectsFiltered = filterData(searchQuery, projects);
 
   let minWidth;
   switch (size) {
-    case 'small':
+    case "small":
       minWidth = 350;
       break;
-    case 'medium':
-    default: 
+    case "medium":
+    default:
       minWidth = 350;
       break;
-    case 'large':
+    case "large":
       minWidth = 300;
   }
 
   return (
-    <Page
-      title="Projects"
-      description="RENCI's projects..."
-    >
+    <Page title="Projects" description="RENCI's projects...">
       <Typography paragraph>
-        Sunt in officia anim aute occaecat sed dolor sit nulla pariatur dolor duis velit veniam.
-        Lorem ipsum cupidatat fugiat adipisicing dolore elit duis reprehenderit anim incididunt in incididunt deserunt sunt culpa fugiat eu minim veniam.
-        Veniam laborum mollit commodo labore excepteur ut sunt dolore dolor in minim enim officia sit occaecat in cillum ullamco.
-        Lorem ipsum ad tempor ea elit anim aliqua nulla exercitation veniam id non elit in in deserunt magna.
+        Sunt in officia anim aute occaecat sed dolor sit nulla pariatur dolor
+        duis velit veniam. Lorem ipsum cupidatat fugiat adipisicing dolore elit
+        duis reprehenderit anim incididunt in incididunt deserunt sunt culpa
+        fugiat eu minim veniam. Veniam laborum mollit commodo labore excepteur
+        ut sunt dolore dolor in minim enim officia sit occaecat in cillum
+        ullamco. Lorem ipsum ad tempor ea elit anim aliqua nulla exercitation
+        veniam id non elit in in deserunt magna.
       </Typography>
 
       <Typography paragraph>
-        Learn more about each project at RENCI below. 
+        Learn more about each project at RENCI below.
       </Typography>
 
-      <Box sx={{
-        flex: 1,
-        marginTop: '3rem',
-        display: 'grid',
-        gap: '2rem',
-        gridTemplateColumns: `repeat(auto-fill, minmax(${minWidth}px, 1fr))`,
-        [theme.breakpoints.down('sm')]: {
-          gridTemplateColumns: '1fr',
-        }
-  
-      }}>
-        {
-          projects.map((project) => {
-            return (
-              <ProjectCard project={project} key={`project-${project.slug}`}/>
-            )
-          })
-        }
+      <SearchBar
+        searchQuery={searchQuery}
+        setSearchQuery={setSearchQuery}
+        options={projectsFiltered}
```

Comment:
I think the autocomplete has the search and filtering functionality built-in to display the list of matching items, so I don't think it makes sense to filter the options? I may be wrong though.